### PR TITLE
[fingerprint] add SourceSkips.PackageJsonScriptsIfNotContainRun

### PR DESCRIPTION
**Pull request by Kudo**
# Why

an inconsistent fingerprint before and after prebuild is that prebuild will change scripts inside package.json. fingerprint includes scripts content and then have different fingerprint. 
https://github.com/expo/expo/blob/d9bdec0a2596437f38584df91133b9c8a37cdf1a/packages/%40expo/cli/src/prebuild/updatePackageJson.ts#L250-L257

# How

introduce `SourceSkips.PackageJsonScriptsIfNotContainRun` and enable this by default

# Test Plan

add unit tests and remove previous updates e2e test workaround

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by wschurman — APPROVED** [review]
One change to make it clearer that this doesn't skip all of scripts, just `android` and `ios`.

File: packages/@expo/fingerprint/src/sourcer/Bare.ts [hunk 1]
```diff
@@ -118,3 +119,19 @@ function stripRncliAutolinkingAbsolutePaths(dependency: any, root: string): void
     }
   }
 }
+
+function normalizePackageJsonScriptSources(
+  scripts: Record<string, string>,
+  options: NormalizedOptions
+): string {
+  if (options.sourceSkips & SourceSkips.PackageJsonScriptsIfNotContainRun) {
+    // Replicate the behavior of `expo prebuild`
```
[the remark this answers is not in the document]
Comment (Kudo) [reply packages/@expo/fingerprint/src/sourcer/Bare.ts:128]: it was kind for the migration from expo-go to development builds. i've heard that we would like to remove the behavior at some point. let's see 👀

**Comment by Kudo** [comment]
just rebase to fix merge conflict

**Comment by Kudo** [comment]
sdk ci failed at unrelated points. going to merge this anyway

**Pull request merged**